src/openrct2-ui/interface/ViewportInteraction.cpp

```diff
@@ -225,40 +224,35 @@ int32_t viewport_interaction_left_click(const ScreenCoordsXY& screenCoords)
  *
  *  rct2: 0x006EDE88
  */
-int32_t viewport_interaction_get_item_right(const ScreenCoordsXY& screenCoords, viewport_interaction_info* info)
+int32_t viewport_interaction_get_item_right(const ScreenCoordsXY& screenCoords, InteractionInfo* info)
 {
     rct_scenery_entry* sceneryEntry;
     Ride* ride;
     int32_t i, stationIndex;
 
     // No click input for title screen or track manager
     if (gScreenFlags & (SCREEN_FLAGS_TITLE_DEMO | SCREEN_FLAGS_TRACK_MANAGER))
-        return info->type = VIEWPORT_INTERACTION_ITEM_NONE;
+        return info->SpriteType = VIEWPORT_INTERACTION_ITEM_NONE;
 
     //
     if ((gScreenFlags & SCREEN_FLAGS_TRACK_DESIGNER) && gS6Info.editor_step != EDITOR_STEP_ROLLERCOASTER_DESIGNER)
-        return info->type = VIEWPORT_INTERACTION_ITEM_NONE;
-
-    auto info2 = get_map_coordinates_from_pos(
-        screenCoords, ~(VIEWPORT_INTERACTION_MASK_TERRAIN & VIEWPORT_INTERACTION_MASK_WATER));
-    info->x = info2.Loc.x;
-    info->y = info2.Loc.y;
-    info->tileElement = info2.Element;
-    info->type = info2.SpriteType;
-    auto tileElement = info->tileElement;
-    auto sprite = info->sprite;
-
-    switch (info->type)
+        return info->SpriteType = VIEWPORT_INTERACTION_ITEM_NONE;
```

Comment:
I know this is how it was, but can you change this into either 
```c++
info->SpriteType == VIEWPORT_INTERACTION_ITEM_NONE
```
(if it was meant to be a comparison)
or
```c++
{
    info->SpriteType = VIEWPORT_INTERACTION_ITEM_NONE;
    return info->SpriteType;
}
```

(Argh, I just noticed it is all over this function)

Reply:
Yeah i was thinking about that. I think I'm going to rejig it to be like that for a different reason. Change coming up shortly